packages/cli/src/commands/update.js

```diff
@@ -0,0 +1,316 @@
+    if (!silent) {
+      console.log(
+        'An update command is already running, please try again in a few seconds.'
+      )
+    }
+    return
+  }
+
+  const updateTasks = new Listr([], {
+    rendererOptions: { collapse: false },
+    renderer: silent ? 'silent' : 'default',
+  })
+
+  // This is only here because we need to display to the user after listr tasks have run
+  // TODO: When listr2 is available handle all output within in the listr2 task then this bool can be removed
+  let upgradeAvailable = false
+
+  updateTasks.add({
+    enabled: () => skip || unskip,
+    title: 'Handling skip flag',
+    task: async (ctx, task) => {
+      const updateData = readUpgradeFile()
+      if (skip) {
+        task.title = `Setting skip version: ${updateData.remoteVersion}`
+        updateData.skipVersion = updateData.remoteVersion
+      } else {
+        task.title = 'Clearing the skip version'
+        updateData.skipVersion = '0.0.0'
+        updateData.lastShown = Date.now() - 2 * getShowPeriod()
+      }
+      writeUpgradeFile(updateData)
+    },
+  })
+
+  updateTasks.add({
+    enabled: () => !(skip || unskip),
+    title: 'Checking for RedwoodJS upgrades',
+    task: async (ctx, task) => {
+      const updateData = await getUpdateVersionStatus()
+      upgradeAvailable = updateData.upgradeAvailable
+      updateData.lastShown = Date.now()
+
+      if (upgradeAvailable) {
+        // reset skip if newer non-skip version is available
+        if (updateData.skipVersion !== updateData.remoteVersion) {
+          updateData.skipVersion = '0.0.0'
+          updateData.lastShown = Date.now() - 2 * getShowPeriod()
+        }
+
+        if (!silent) {
+          // TODO: When listr2: prompt if they want to upgrade now? Can add on exit hook to run `yarn rw upgrade`
+        }
+
+        task.title =
+          updateData.skipVersion !== updateData.remoteVersion
+            ? `New upgrade is available: ${updateData.remoteVersion}`
+            : `New upgrade is available: ${updateData.remoteVersion} (you are currently skipping this upgrade)`
+      } else {
+        task.title = 'No upgrade is available'
+      }
+
+      writeUpgradeFile(updateData)
+    },
+  })
+
+  try {
+    setLock(UPDATE_LOCK)
+    await updateTasks.run()
+    if (upgradeAvailable && !silent) {
+      showUpgradeAvailableMessage()
+    }
+    unsetLock(UPDATE_LOCK)
+  } catch (e) {
+    unsetLock(UPDATE_LOCK)
+    errorTelemetry(process.argv, e.message)
+    console.error(c.error(e.message))
+    process.exit(e?.exitCode || 1)
+  }
+}
+
+async function getUpdateVersionStatus() {
+  // Read package.json and extract the @redwood/core version
+  const packageJson = JSON.parse(
+    fs.readFileSync(path.join(getPaths().base, 'package.json'))
+  )
+  let localVersion = packageJson.devDependencies['@redwoodjs/core']
+
+  // Remove any leading non-digits, i.e. ^ or ~
+  while (!(localVersion.charAt(0) >= '0' && localVersion.charAt(0) <= '9')) {
```

Comment:
Could make this a little less verbose with a regex:
```suggestion
  while (!/\d/.test(localVersion.charAt(0))) {
```

Reply:
Okay this is much less verbose. I updated to use this.